controllers/etcd_custodian_controller.go

```diff
@@ -80,12 +125,183 @@ func (ec *EtcdCustodian) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.
+					BlockOwnerDeletion: pointer.BoolPtr(true),
+					Controller:         pointer.BoolPtr(true),
+					Kind:               "Etcd",
+					Name:               etcd.Name,
+					UID:                etcd.UID,
+				},
+			}
+			return nil
+		})
+		if err != nil {
+			logger.Error(err, "Full snap lease found but error occured during claiming it")
+			return ctrl.Result{
+				RequeueAfter: 5 * time.Second,
+			}, nil
+		}
+	}
+
+	// Get or Create delta_snapshot_revisions lease object that will keep track of delta snapshot revisions based on which
+	// compaction job will be scheduled
+	nsName = types.NamespacedName{
+		Name:      "delta-snapshot-revisions",
+		Namespace: etcd.Namespace,
+	}
+
+	deltaLease := &coordinationv1.Lease{}
+	err1 = ec.Get(ctx, nsName, deltaLease)
+	if err1 != nil {
+		logger.Info("Couldn't fetch delta snap lease delta-snapshot-revisions because: " + err1.Error())
+		logger.Info("Creating the delta snap lease delta-snapshot-revisions")
+
+		renewTime := metav1.NewMicroTime(time.Now())
+		deltaLease = &coordinationv1.Lease{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "delta-snapshot-revisions",
+				Namespace: etcd.Namespace,
+				OwnerReferences: []metav1.OwnerReference{
+					{
+						APIVersion:         "druid.gardener.cloud/v1alpha1",
+						BlockOwnerDeletion: pointer.BoolPtr(true),
+						Controller:         pointer.BoolPtr(true),
+						Kind:               "Etcd",
+						Name:               etcd.Name,
+						UID:                etcd.UID,
+					},
+				},
+			},
+			Spec: coordinationv1.LeaseSpec{
+				HolderIdentity: pointer.StringPtr("0"),
+				RenewTime:      &renewTime,
+			},
+		}
+		err2 := ec.Create(ctx, deltaLease)
+		if err2 != nil {
+			logger.Error(err2, "Delta snap lease delta-snapshot-revisions couldn't be created")
+			return ctrl.Result{
+				RequeueAfter: 5 * time.Second,
+			}, nil
+		}
+
+		deltaLease = &coordinationv1.Lease{}
+		err2 = ec.Get(ctx, nsName, deltaLease)
+		if err2 != nil {
+			logger.Error(err2, "Delta snap lease delta-snapshot-revisions couldn't be fetched just after creation")
+			return ctrl.Result{
+				RequeueAfter: 5 * time.Second,
+			}, nil
+		}
+	}
+
+	if !checkEtcdOwnerReference(deltaLease.GetOwnerReferences(), etcd) {
+		err := kutil.TryPatch(ctx, retry.DefaultBackoff, ec.Client, deltaLease, func() error {
+			deltaLease.OwnerReferences = []metav1.OwnerReference{
+				{
+					APIVersion:         "druid.gardener.cloud/v1alpha1",
+					BlockOwnerDeletion: pointer.BoolPtr(true),
+					Controller:         pointer.BoolPtr(true),
+					Kind:               "Etcd",
+					Name:               etcd.Name,
+					UID:                etcd.UID,
+				},
+			}
+			return nil
+		})
+		if err != nil {
+			logger.Error(err, "Delta snap lease found but error occured during claiming it")
+			return ctrl.Result{
+				RequeueAfter: 5 * time.Second,
+			}, nil
+		}
+	}
+	// Run compaction job
+	if etcd.Spec.Backup.Store != nil {
+		hi, err := strconv.ParseInt(*deltaLease.Spec.HolderIdentity, 10, 32)
```

Comment:
Can you elaborate what component resets this count after the compaction was successful?

Reply:
That component is present in etcd br. After etcd br executes compaction, this value will be reset